backend/app/adapter/routing/handle/search.go

```diff
@@ -205,3 +211,24 @@ func newUser(user entity.User) User {
 		UpdatedAt:      user.UpdatedAt,
 	}
 }
+
+func emitSearchError(w http.ResponseWriter, err error) {
+	var (
+		code = http.StatusInternalServerError
+		u    search.ErrUserNotProvided
+		r    search.ErrUnknownResource
+	)
+	if errors.As(err, &u) {
+		code = http.StatusUnauthorized
+	}
+	if errors.As(err, &r) {
+		code = http.StatusNotFound
+	}
+	errResp, err := json.Marshal(SearchError{
+		Message: err.Error(),
+	})
+	if err != nil {
+		return
+	}
+	http.Error(w, string(errResp), code)
+}
```

Comment:
Each search error is different in the code. I recommend putting the content of this function back to the code so that the reader can follow the error handling logic. This is an example when abstraction is reducing readability.